src/ObjLoading/Game/IW4/Leaderboard/JsonLeaderboardDefLoader.cpp

```diff
@@ -0,0 +1,110 @@
+#include "JsonLeaderboardDefLoader.h"
+
+#include "Game/IW4/CommonIW4.h"
+#include "Game/IW4/Leaderboard/JsonLeaderboardDef.h"
+
+#include <iostream>
+#include <nlohmann/json.hpp>
+
+using namespace nlohmann;
+using namespace IW4;
+
+namespace
+{
+    class JsonLoader
+    {
+    public:
+        JsonLoader(std::istream& stream, MemoryManager& memory)
+            : m_stream(stream),
+              m_memory(memory)
+        {
+        }
+
+        bool Load(LeaderboardDef& leaderboardDef) const
+        {
+            const auto jRoot = json::parse(m_stream);
+            std::string type;
+            unsigned version;
+
+            jRoot.at("_type").get_to(type);
+            jRoot.at("_version").get_to(version);
+
+            if (type != "leaderboard" || version != LEADERBOARD_DEF_VERSION)
+            {
+                std::cerr << "Tried to load leaderboard \"" << leaderboardDef.name << "\" but did not find expected type leaderboard of version "
+                          << LEADERBOARD_DEF_VERSION << "\n";
+                return false;
+            }
+
+            const auto jLeaderboard = jRoot.get<JsonLeaderboardDef>();
+            return CreateLeaderboardFromJson(jLeaderboard, leaderboardDef);
+        }
+
+    private:
+        bool CreateColumnDefFromJson(const JsonColumnDef& jColumn, LbColumnDef& lbColumnDef, LeaderboardDef& leaderboardDef) const
+        {
+            lbColumnDef.name = m_memory.Dup(jColumn.name.c_str());
+
+            if (jColumn.propertyId)
+                lbColumnDef.propertyId = jColumn.propertyId.value();
+
+            if (jColumn.hidden)
+                lbColumnDef.hidden = jColumn.hidden.value();
+
+            if (jColumn.statName)
+                lbColumnDef.statName = m_memory.Dup(jColumn.statName.value().c_str());
```

Comment:
```suggestion
                lbColumnDef.statName = m_memory.Dup(jColumn.statName->c_str());
```

You can use `->` for std::optional, which does the same as `.value()`